firmware/controllers/lua/lua_hooks.cpp

```diff
@@ -394,6 +395,45 @@ static int lua_setAirmass(lua_State* l) {
 	return 0;
 }
 
+auto& checkBackupSram(lua_State* l) {
+	auto backupRam = getBackupSram();
+
+	if (!backupRam) {
+		luaL_error(l, "Backup RAM error or not supported!");
+	}
+
+	return *backupRam;
+}
+
+static int lua_getPersistentValue(lua_State* l) {
+	auto idx = luaL_checkinteger(l, 1);
+
+	auto backupRam = checkBackupSram(l);
+
+	if (idx < 0 || idx > efi::size(backupRam.LuaPersistentData)) {
```

Comment:
perform 0- vs 1- based index translation for the Lua scripter, as we do for other Lua APIs

also, bug here about `idx == 64` (`efi::size(backupRam.LuaPersistentData)`)

```suggestion
	if (idx < 1 || idx > efi::size(backupRam.LuaPersistentData)) {
		luaL_error(l, "invalid backup ram index: %d", idx);
	}
	
	lua_pushnumber(l, backupRam.LuaPersistentData[idx-1]);
```